framework/encode/vulkan_capture_manager.cpp

```diff
@@ -2822,5 +2823,34 @@ void VulkanCaptureManager::PreProcess_vkBindImageMemory2(VkDevice
     }
 }
 
+void VulkanCaptureManager::PostProcess_vkCreateShaderModule(VkResult                        result,
+                                                            VkDevice                        device,
+                                                            const VkShaderModuleCreateInfo* pCreateInfo,
+                                                            const VkAllocationCallbacks*    pAllocator,
+                                                            VkShaderModule*                 pShaderModule)
+{
+    GFXRECON_UNREFERENCED_PARAMETER(device);
+    GFXRECON_UNREFERENCED_PARAMETER(pAllocator);
+    GFXRECON_UNREFERENCED_PARAMETER(pShaderModule);
+
+    if (result == VK_SUCCESS)
+    {
+        // spirv-parsing for buffer-references
+        gfxrecon::util::SpirVParsingUtil spirv_util;
+
+        if (spirv_util.ParseBufferReferences(pCreateInfo->pCode, pCreateInfo->codeSize))
+        {
+            auto buffer_reference_infos = spirv_util.GetBufferReferenceInfos();
+
+            if (!buffer_reference_infos.empty())
+            {
+                GFXRECON_LOG_WARNING_ONCE(
```

Comment:
If the buffer device capture replay feature is enabled will SPIRV buffer references still be a problem for GFXR, or can the warning be skipped?

Reply:
as I understand the feature, it would give you the same addresses for consecutive runs of the same application, on the same device. 
_"indicates that the implementation supports saving and reusing buffer and device addresses"._

we still need to do the detection in spirv, later extract the addresses from buffers/copies, map+track the pointed-to buffers, etc.
I cannot see how "device capture replay" could help or solve that on its own, esp. when replaying on other driver/devices.
so I think we should issue this warning also with capture-replay being available (and maybe re-visit the other warning as I'm unclear what exactly we'd gain from having the feature available)

Reply:
My understanding was that BDAs need to be identified in shaders so those BDAs can be translated from capture to replay time values. If capture and replay is done on the same device/driver using the BDA capture replay feature then capture-time and replay-time BDA values should be the same and no translating is needed. Is there other mapping or tracking that needs to be done for GFXR to support this use case?

I agree translation is needed for future support of capture and replay on different devices/drivers--and also may be needed to support devices/drivers that don't support the BDA capture replay feature.

Reply:
_Is there other mapping or tracking that needs to be done for GFXR to support this use case?_
in my test-cases so far, using ubuntu/x86_64/nvidia, I can confirm this is like you point out. samples using buffer-references seem to 'magically' capture/replay fine locally. trimming would be a case I 'think' we would currently hit a wall. we do not track buffer-references as pipeline-dependencies, so would potentially trim them away.